homeassistant/components/lutron_caseta/button.py

```diff
@@ -0,0 +1,72 @@
+"""Support for pico and keypad buttons."""
+from __future__ import annotations
+
+from homeassistant.components.button import ButtonEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import LutronCasetaDevice
+from .const import DOMAIN as CASETA_DOMAIN
+from .device_trigger import (
+    LEAP_TO_DEVICE_TYPE_SUBTYPE_MAP,
+    _lutron_model_to_device_type,
+)
+from .models import LutronCasetaData
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Lutron pico and keypad buttons."""
+    data: LutronCasetaData = hass.data[CASETA_DOMAIN][config_entry.entry_id]
+    bridge = data.bridge
+    bridge_devices = bridge.get_devices()
+    button_devices = bridge.get_buttons()
+
+    for button_device_id in button_devices:
+        # find and complete missing button names
+        button_device = button_devices[button_device_id]
+        ha_device = button_device
+        if (
+            "parent_device" in button_device
+            and button_device["parent_device"] is not None
+        ):
+            ha_device = bridge_devices[button_device["parent_device"]]
+
+        if "device_name" not in button_device:
+            # device name (button name) is missing, probably a caseta pico
+            # try to get the name using the button number from the triggers
+            button_numbers = LEAP_TO_DEVICE_TYPE_SUBTYPE_MAP.get(
+                _lutron_model_to_device_type(ha_device["model"], ha_device["type"]),
+                {},
+            )
+            button_device["device_name"] = button_numbers.get(
+                int(button_device["button_number"]),
+                f"button {button_device['button_number']}",
+            )
```

Comment:
Caseta does not send the name of the buttons for Picos, just the button number.  This looks up the button name from the device_triggers.

The bigger question though, is do we want to expose buttons on caseta systems?  
Will a caseta user ever care to push a pico button from within HA?
RA3/HWQSX users need this to trigger scenes, but I'm not sure the same need exists with caseta.

We could simply not create entities for caseta buttons.  

I'm not sure what the right path is.

Reply:
@bdraco, I'm curious about your opinion on whether or not to create button entities for caseta pico buttons, or to exclude them.

Reply:
I think its fine to create them but they probably should be disabled by default with `entity_registry_enabled_default=False`

Reply:
great idea.  I've implemented this